r/tests/testthat/test-dataset.R

```diff
@@ -618,6 +618,33 @@ test_that("UnionDataset handles InMemoryDatasets", {
   expect_equal(actual, expected)
 })
 
+test_that("scalar aggregates with many batches", {
+  test_data <- data.frame(val = 1:1e7)
+  expected_result_distr <- (
+    sapply(1:100, function(iter_ndx) {
+      test_data                              %>%
+        dplyr::summarise(min_val = min(val)) %>%
+        dplyr::collect()                     %>%
+        dplyr::pull(min_val)
+    }) %>%
+      table()
+  )
+
+  ds_tmpfile <- tempfile("test-aggregate", fileext = ".parquet")
+  arrow::write_parquet(test_data, ds_tmpfile)
+  actual_result_distr <- (
+    sapply(1:100, function(iter_ndx) {
+      arrow::open_dataset(ds_tmpfile)        %>%
+        dplyr::summarise(min_val = min(val)) %>%
+        dplyr::collect()                     %>%
+        dplyr::pull(min_val)
+    }) %>%
+      table()
+  )
```

Comment:
nit: if we aren't using iter_idx, can use replicate
```suggestion
  actual_result_distr <- (
    replicate(100, {
      arrow::open_dataset(ds_tmpfile)        %>%
        dplyr::summarise(min_val = min(val)) %>%
        dplyr::collect()                     %>%
        dplyr::pull(min_val)
    }) %>%
      table()
  )
```

Reply:
Nice! Neal's suggestion included this change, so I took that.